src/edfi-paging-test/edfi_paging_test/api/request_client.py

```diff
@@ -215,15 +190,15 @@ def get_all(
             A list of all parsed results
         """
 
-        if page_size is None:
-            page_size = self.page_size
-
-        pagination_result = self.get_page(1, resource, page_size)
+        pagination_result = self.get_page(resource, 1)
 
         items: List[Any] = []
         while True:
-            items = items + list(pagination_result.current_page_items)
-            pagination_result = self.get_page(pagination_result.current_page + 1)
+            items.extend(pagination_result.current_page_items)
```

Comment:
I didn't notice before that we had `items + list()` - that's actually a dangerous pattern in Python. Better to use `extend()`.